apps/rush-lib/src/cli/actions/PublishAction.ts

```diff
@@ -346,6 +358,10 @@ export class PublishAction extends BaseRushAction {
         args.push(`--force`);
       }
 
+      if (this._npmAccessLevel.value) {
+        args.push(`--access`, this._npmAccessLevel.value);
```

Comment:
Do we want to validate the specified value?

Reply:
We can, if an invalid value is specified the package manager will simply warn and use its default behavior:

```shell
$ npm publish --access foo
npm WARN invalid config access="foo"
+ @kevintcoughlin/test-rush-publish@1.0.0
```

Reply:
*"Silently do the wrong thing"* is NPM's motto, not Rush's.  :-P

Reply:
That said, it looks like your `defineChoiceParameter()` already constrains the allowable values.

Reply:
> s

Indeed:

```shell
rush publish: error: argument "--set-access-level": Invalid choice: foo (choose from [public, restricted])
```

Reply:
Correct. Although is it safe to rely on this? What is someone changes this to a string value one day? Or adds another value that NPM doesnt recognize?

---
In reply to: [256626767](https://github.com/Microsoft/web-build-tools/pull/1019#discussion_r256626767) [](ancestors = 256626767)